core/controllers/editor.py

```diff
@@ -854,14 +858,23 @@ def post(self, exploration_id):
                 'Expected a filename ending in .%s, received %s' %
                 (file_format, filename))
 
-        # Save the file.
-        fs = fs_domain.AbstractFileSystem(
-            fs_domain.ExplorationFileSystem(exploration_id))
-        if fs.isfile(filename):
+        mimetype = '%s/%s' % (self._FILENAME_PREFIX, extension)
```

Comment:
Use just `'image/%s' % (extension)` here, bacause the constant has now two meanings one is path to images and the second is part of mimetype.

Reply:
Done!